### PR TITLE
Adds min reqs.

### LINKED ISSUE #127: Call our minimum requirements for client library
We should call out requirements for the client library on the landing page of the reference: https://docs.web3.storage/reference/client-library/

We may also want to call out some requirements for common environments, eg webpack. see: https://github.com/web3-storage/web3.storage/issues/260

### PR DESCRIPTION
**Pull request by johnnymatthews**
Closes #127

image

### CODE REVIEW

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!

### `docs/community/help-and-support.md`:
- The following link returned HTTP status code 302 (Found):
  https://filecoin.io/slack
- The following link returned HTTP status code 302 (Found):
  https://github.com/web3-storage/web3.storage/issues/new/choose
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... channel on Slack.  (opens new window)  Found a bug, or have a feature reque...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...eb3.Storage service  (opens new window) File an issue for the Web3.Storage d...
                                             ^^^^
  ```
- Consider using “email”.
  ```
  ...  Need more space on Web3.Storage? Just send an email to request more. The W...
                                             ^^^^^^^^^^^^^^^^
  ```

### `docs/how-tos/retrieve.md`:
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...is guide, we'll use the gateway at dweb.link, but you can see more worldwide...
                                             ^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ...uy7pcu, you can make a URL for the dweb.link gateway as follows: bafybeidd2g...
                                             ^^^^
  ```
- Use ‘own’ only if you want to emphasize that something belongs to a person or thing or was done by a person or thing. Consider writing “your”.
  ```
  ... so if you're looking to link to one of your own files, you don't even have ...
                                             ^^^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... you have a client, you can call client.get, passing in a CID string:  The g...
                                             ^^^
  ```
- An apostrophe may be missing.
  ```
  ...S data: files and unixFsIterator(). The files method returns an array of Web...
                                             ^^^^^
  ```
- In American English, the usual prepositions after ‘different’ are “from” or “than”.
  ```
  ...save the file to. This can be different to <FILE NAME>. Desktop/output-file....
                                             ^^
  ```
- In American English, the usual prepositions after ‘different’ are “from” or “than”.
  ```
  ...save the file to. This can be different to <FILE NAME>. Desktop/output-file....
                                             ^^
  ```
- In American English, the usual prepositions after ‘different’ are “from” or “than”.
  ```
  ...save the file to. This can be different to <FILE NAME>. Desktop/output-file....
                                             ^^
  ```
- Possibly missing comma found.
  ```
  ...ent to <FILE NAME>. Desktop/output-file.txt Your complete command should loo...
                                             ^^^
  ```
- The verb “check out” can sound colloquial in this context. Consider using a more formal alternative.
  ```
  ...h how to store data using Web3.Storage, check out the storage how-to guide f...
                                             ^^^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ... can also use the client library to get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/how-tos/store.md`:
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...s such as fetch  (opens new window) and File  (opens new window). This guide...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ....storage package to your dependencies:  First we need to create a Web3.Stora...
                                             ^^^^^
  ```
- The phrase ‘a lot of’ might be wordy. Consider using “many”.
  ```
  ...ent method: If you expect to be loading a lot of large files, you may be bet...
                                             ^^^^^^^^
  ```
- Apostrophe might be missing. Did you mean “file’s” (curly possessive apostrophe) or “file's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...one content archive, and linking to the files inside is much easier if each ...
                                             ^^^^^
  ```
- Possibly missing comma found.
  ```
  ...an array of Files, uploading is simple: IMPORTANT Deleting files from the We...
                                             ^^^^^^^^^
  ```
- Possibly missing article found.
  ```
  ... IMPORTANT Deleting files from the Web3.Storage site's Files page  (opens ne...
                                             ^^^^^^^
  ```
- Consider using a more formal alternative to avoid the overused word “more”.
  ```
  ...to guide on retrieval. You can also get more information about the status of...
                                             ^^^^^^^^^^^^^^^^
  ```
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘your’ (belonging to you) seems less likely than ‘you’ (personal pronoun).
  ```
  ...ties that can be used as parameters for your put() operation: name String. T...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...rameters for your put() operation: name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...ens new window) format. pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```

File: docs/reference/client-library.md [hunk 1]
```diff
@@ -9,6 +9,14 @@ To use the JavaScript client library for Web3.Storage, you must first [obtain a
 
 The client library automatically packs your uploads into a content addressible archive (CAR) for uploading to the Web3.Storage service, which [stores](#store-files) data as blocks prefixed with the [_content identifier_ (CID)](../concepts/content-addressing.md#cids-location-independent-globally-unique-keys) derived from a cryptographic hash of the data. You can then use a file's CID to [retrieve](#retrieve-files) it.
 
+:::warning Minimum requirements
+While we recommend that you install the latest _stable_ version of the following software, you must have _at least_:
+
+- [NPM](https://www.npmjs.com/) `7.0.0`
+- [Nodejs](https://nodejs.org/en/) `14.0.0`
```
Comment (jessicaschilling) [inline docs/reference/client-library.md:16]:
```suggestion
- [Node.js](https://nodejs.org/en/) `14.0.0`
```

**Review by jessicaschilling — CHANGES_REQUESTED** [review]
Minor consistency fix, otherwise LGTM!

File: docs/reference/client-library.md (hunk 1 above)
Comment (johnnymatthews) [reply docs/reference/client-library.md:16]: Still waiting on the day that someone decides to make `.js` a TLD...

Comment (jessicaschilling) [reply docs/reference/client-library.md:16]: I have The Fear.

**Comment by filecorgi** [comment]
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/reference/client-library.md`:
- Possibly missing preposition found.
  ```
  ... the JavaScript client library for Web3.Storage, you must first obtain a fre...
                                             ^^^^^^^
  ```
- ‘put’ seems less likely than ‘out’.
  ```
  ... can be used as parameters when calling put(): name String. The name paramet...
                                             ^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ... as parameters when calling put(): name String. The name parameter lets you ...
                                             ^^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...r IPFS path. For example, if you have a files directory like this: Using the...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...turn a CID for a directory containing a files subdirectory, like this: Howev...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ou do this instead: The contents of the files directory will be at the top l...
                                             ^^^^^
  ```
- An apostrophe may be missing.
  ```
  ...ill be at the top level, instead of the files directory itself:  onRootCidRe...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.
  ```
  ... the iterator represent files. If entry.type == 'directory', the entry repre...
                                             ^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Possibly missing comma found.
  ```
  ...tional order. Number Type Description 1 string A string containing the CID o...
                                             ^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...ow) containing the queried CID. created String. The created property gives t...
                                             ^^^^^^
  ```
- Apostrophe might be missing. Did you mean “pin’s” (curly possessive apostrophe) or “pin's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ...pload the same file(s). pins Array. The pins property is an array of Pin obj...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...operty is an array of Pin objects. Each Pin object represents a specific pee...
                                             ^^^
  ```
- Apostrophe might be missing. Did you mean “deal’s” (curly possessive apostrophe) or “deal's” (straight possessive apostrophe)? (Some terms such as “farmers’ market” have a possessive apostrophe after the first noun.)
  ```
  ... following structure:  deals Array. The deals property is an array of Deal o...
                                             ^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...perty is an array of Deal objects. Each Deal object represents a specific st...
                                             ^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).
  ```
  ...t with the following properties: before String. Specifies a date, in ISO 860...
                                             ^^^^^^
  ```

**Comment by yusefnapora** [comment]
@johnnymatthews I noticed a little conflict with main after I merged in #133, so I pushed up a merge commit. Just FYI :)

**Review by jessicaschilling — APPROVED** [review]

**Review by atopal — COMMENTED** [review]
LGTM!

**Pull request merged**